document/CoreFormat.mdk

```diff
@@ -839,70 +832,61 @@ This allows tools to include information about the themselves that is not explic
 
 ### General { #file-General }
 
-A `file` object represents a single file. In some cases, this file might be nested within another file,
-referred to as the "containing file."
-
-`file` objects occur as members of arrays within the `run.files` object (&sect;[#run-files]). 
-
-If the `file` object does not represent a nested file, neither of the properties
-`uri` (&sect;[#file-uri]) or `offset` (&sect;[#file-offset]) shall be present.
-
-If the location of a nested file can be expressed either by means of a path within the containing file
-or by means of a byte offset from the start of the containing file, then either `uri`
-or `offset` (&sect;[#file-offset]) or both shall be present; they shall not both be absent.
+A `file` object represents a single file.
 
 ### `uri` property { #file-uri }
 
-Depending on the circumstances, a `file` object either shall or may contain a property named `uri`
-whose value is a string.
+Depending on the circumstances, a `file` object either shall, may, or shall not contain a property named `uri`
+whose value is a string containing a valid URI.
 
-If the `file` object represents a top-level file (that is, a file that is not nested within another file),
-then the `uri` property may be present.
+If the `file` object represents a top-level file, then the `uri` property may be present.
 If present, it shall be equal to the name of the property within `run.files` (&sect;[#run-files])
-within which the `file` object occurs, but with any URI fragment removed.
-If absent, it shall be interpreted as being equal to the value it would be required to have if it were present.
+whose value is this `file` object.
+If absent, it shall be interpreted as having that same value.
 
-If the `file` object represents a nested file, and if the location of the nested file
-within its containing file can be expressed by means of a path within its containing file,
+If the `file` object represents a nested file whose location
+relative to its parent can be expressed only by means of a path,
 then the `uri` property shall be present, and its value shall be a valid relative URI expressing that path.
 
-If the location of the nested file within its containing file cannot be expressed by means of a path,
+If the `file` object represents a nested file whose location within its parent cannot be expressed by means of a path,
```

Comment:
i see you make this all clear in the following text, just a parenthetical aside might be useful here, because it is unclear at this point in the text.

---
In reply to: [64078552](https://github.com/sarif-standard/sarif-spec/pull/209#discussion_r64078552) [](ancestors = 64078552)